includes/validation/class-amp-validation-manager.php

```diff
@@ -1975,32 +1956,72 @@ public static function get_validate_url_error_message( $error_code, $error_messa
 			);
 		}
 
+		$implode_non_empty_strings_with_spaces = static function ( $strings ) {
+			return implode( ' ', array_filter( $strings ) );
+		};
+
 		switch ( $error_code ) {
 			case 'http_request_failed':
-				return __( 'Failed to fetch URL to validate.', 'amp' ) . $error_message . $site_health_message . $support_forum_message;
+				return $implode_non_empty_strings_with_spaces(
+					[
+						esc_html__( 'Failed to fetch URL to validate.', 'amp' ),
+						esc_html( $error_message ),
+						$site_health_message,
+						$support_forum_message,
+					]
+				);
 			case 'white_screen_of_death':
-				return __( 'Unable to validate URL. Encountered a white screen of death likely due to a PHP fatal error.', 'amp' ) . $error_message . $check_error_log . $support_forum_message;
+				return $implode_non_empty_strings_with_spaces(
+					[
+						esc_html__( 'Unable to validate URL. A white screen of death was encountered which is likely due to a PHP fatal error.', 'amp' ),
+						esc_html( $error_message ),
+						$check_error_log,
+						$support_forum_message,
+					]
+				);
 			case '404':
-				return __( 'The fetched URL was not found. It may have been deleted. If so, you can trash this.', 'amp' ) . $error_message . $support_forum_message;
+				return $implode_non_empty_strings_with_spaces(
+					[
+						esc_html__( 'The fetched URL was not found. It may have been deleted. If so, you can trash this.', 'amp' ),
+						esc_html( $error_message ),
+						$support_forum_message,
+					]
+				);
 			case '500':
-				return __( 'An internal server error occurred when fetching the URL for validation.', 'amp' ) . $error_message . $check_error_log . $support_forum_message;
-			case 'response_comment_absent':
-				return sprintf(
-					/* translators: %1$s: AMP_VALIDATION, %2$s: </body> */
-					__( 'URL validation failed to due to the absence of the expected JSON-containing %1$s HTML comment after %2$s. This is often due to a PHP fatal error occurring.', 'amp' ),
-					'<code>AMP_VALIDATION</code>',
-					'<code>&lt;/body&gt;</code>'
-				) . $error_message . $check_error_log . $support_forum_message;
+				return $implode_non_empty_strings_with_spaces(
+					[
+						esc_html__( 'An internal server error occurred when fetching the URL for validation.', 'amp' ),
+						esc_html( $error_message ),
+						$check_error_log,
+						$support_forum_message,
+					]
+				);
+			case 'response_not_json':
+				return $implode_non_empty_strings_with_spaces(
+					[
+						esc_html__( 'URL validation failed to due to the AMP validation request not returning JSON data. This is may be due to a PHP fatal error occurring.', 'amp' ),
```

Comment:
```suggestion
						esc_html__( 'URL validation failed due to the AMP validation request not returning JSON data. This may be due to a PHP fatal error occurring.', 'amp' ),
```